pages/scenario_selection_page.py

```diff
@@ -309,67 +381,344 @@ def scenario_selection_page():
+        fuel_type="gas",
+        wholesale_price_projection_scenario=wholesale_price_projection,
+        include_standing_charge=True,  # in phase 1 of the project, we decided that the gas standing charge should only be included in the running costs of a gas boiler
+        levy_rebalancing=levy_rebalancing,
+        levies_to_rebalance=levies_to_rebalance,
+        levy_rebalancing_weights=levies_rebalancing_weights,
+        include_vat=True,
+    )
+
+    boiler_lifetime_costs = cost_calculator.compute_total_lifetime_costs(
+        installation_costs=boiler_upfront_costs,
+        maintenance_costs=boiler_maintenance_costs,
+        running_costs=boiler_running_costs,
+    )
+
+    ashp_lifetime_costs["technology"] = "ASHP"
+    ashp_lifetime_costs["life_span"] = ashp_life_span
+    boiler_lifetime_costs["technology"] = "Gas boiler"
+    boiler_lifetime_costs["life_span"] = boiler_life_span
+
+    # Merging ASHP and boiler results
+    lifetime_costs = pd.concat(
+        [ashp_lifetime_costs, boiler_lifetime_costs]
+    ).reset_index()
+
+    lifetime_costs["installation_costs_after_subsidy"] = (
+        lifetime_costs["installation_costs"]
+        - lifetime_costs["subsidy_value"]
+    )
+    lifetime_costs.drop(
+        columns=["installation_costs", "subsidy_value"], inplace=True
+    )
+
+    lifetime_costs["archetype_label"] = lifetime_costs["archetype_label"].map(
+        archetype_name_mapping
+    )
+
+    # needs to be QAed
+    number_of_properties = {'Post-1950 semi/terraced house': 6841365,
+    'Pre-1950 semi/terraced house': 5602441,
+    'Post-1950 flat': 4076569,
+    'Post-1950 detached house': 3103702,
+    'Post-1950 bungalow': 1564001,
+    'Pre-1950 flat': 1415792,
+    'Pre-1950 detached house': 1059425,
+    'Pre-1950 bungalow': 205582}
+
+    lifetime_costs["number_of_properties"] = lifetime_costs["archetype_label"].map(number_of_properties)
+    # Adding weighted average archetype
+    weighted_lifetime_costs = lifetime_costs.groupby("technology").apply(
+        lambda x: pd.Series(
+            {
+                "lifetime_running_costs": (x["lifetime_running_costs"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "installation_costs_after_subsidy": (x["installation_costs_after_subsidy"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "loan_interest": (x["loan_interest"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "lifetime_maintenance_costs": (x["lifetime_maintenance_costs"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "number_of_properties": x["number_of_properties"].sum(),
+            }
+        )
+    ).reset_index()
+    weighted_lifetime_costs["archetype_label"] = "Weighted average archetype"
+    weighted_lifetime_costs["life_span"] = (
+        lifetime_costs["life_span"]
+        .mean()
+        .round()
+        .astype(int)
+    )
```

Comment:
```suggestion
    life_span_map = {"ASHP": ashp_life_span,
    "Gas boiler": boiler_life_span}
    weighted_lifetime_costs["life_span"] = weighted_lifetime_costs["technology"].map(life_span_map)
```
This is very minor and doesn't change the outcome but think it'd be clearer to use the `ashp_life_span` and `boiler_life_span` variables directly for the lifespan for the weighted average archetype rows